test/partitioned_databases.js

```diff
@@ -0,0 +1,230 @@
+// Copyright © 2019 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it before after */
+'use strict';
+
+const assert = require('assert');
+const async = require('async');
+const Cloudant = require('../cloudant.js');
+const nock = require('./nock.js');
+const uuidv4 = require('uuid/v4'); // random
+
+const ME = process.env.cloudant_username || 'nodejs';
+const PASSWORD = process.env.cloudant_password || 'sjedon';
+const SERVER = process.env.SERVER_URL || `https://${ME}.cloudant.com`;
+const DBNAME = `nodejs-cloudant-${uuidv4()}`;
+
+describe('Partitioned Databases #db', function() {
+  const partitionKeys = Array.apply(null, {length: 10})
+        .map(() => { return uuidv4(); });
+
+  before(function(done) {
+    var mocks = nock(SERVER)
+      .put(`/${DBNAME}`)
+      .query({ partitioned: true })
+      .reply(201, { ok: true });
+
+    const cloudant = Cloudant({ url: SERVER, username: ME, password: PASSWORD });
+    cloudant.db.create(DBNAME, { partitioned: true }).then((body) => {
```

Comment:
The test style is fine, but if we wanted to make it more concise now that we have promises I think we can just return the promise to mocha i.e.
```js
return cloudant.db.create(DBNAME, { partitioned: true }).then((body) => {
      assert.ok(body.ok);
      mocks.done();
    })
```